service/controller/v24/controllercontext/cluster_asg.go

```diff
@@ -1,11 +1,12 @@
 package controllercontext
 
-type ClusterASG struct {
+type ContextStatusTenantClusterTCCPASG struct {
```

Comment:
The names do not get prettier but this is just how it is I guess.

Reply:
Oof that is touch to read here. But when its used `cc.Status.TenantCluster.TCCP.ASG.Name` I'm fine with it.